Source/Scene/Model.js

```diff
@@ -1135,9 +1149,86 @@ define([
                 if (value === lightColor || Cartesian3.equals(value, lightColor)) {
                     return;
                 }
-                this._regenerateShaders = this._regenerateShaders || (defined(lightColor) && !defined(value)) || (defined(value) && !defined(lightColor));
+                this._shouldRegenerateShaders = this._shouldRegenerateShaders || (defined(lightColor) && !defined(value)) || (defined(value) && !defined(lightColor));
                 this._lightColor = Cartesian3.clone(value, lightColor);
             }
+        },
+
+        /**
+         * The sun's luminance at the zenith in kilo candela per meter squared to use for this model's procedural environment map.
+         * This is used when {@link Model#specularEnvironmentMaps} and {@link Model#sphericalHarmonicCoefficients} are not defined.
+         *
+         * @memberof Model.prototype
+         *
+         * @demo {@link https://cesiumjs.org/Cesium/Apps/Sandcastle/index.html?src=Image-Based Lighting.html|Sandcastle Image Based Lighting Demo}
+         * @type {Number}
+         * @default 1.0
+         */
+        luminanceAtZenith : {
+            get : function() {
+                return this._luminanceAtZenith;
+            },
+            set : function(value) {
+                var lum = this._luminanceAtZenith;
+                if (value === lum) {
+                    return;
+                }
+                this._shouldRegenerateShaders = this._shouldRegenerateShaders || (defined(lum) && !defined(value)) || (defined(value) && !defined(lum));
+                this._luminanceAtZenith = value;
+            }
+        },
+
+        /**
+         * The third order spherical harmonic coefficients used for the diffuse color of image-based lighting. When <code>undefined</code>, a diffuse irradiance
+         * computed from the atmosphere color is used.
+         * <p>
+         * There are nine <code>Cartesian3</code> coefficients.
+         * The order of the coefficients is: L<sub>00</sub>, L<sub>1-1</sub>, L<sub>10</sub>, L<sub>11</sub>, L<sub>2-2</sub>, L<sub>2-1</sub>, L<sub>20</sub>, L<sub>21</sub>, L<sub>22</sub>
+         * </p>
+         *
+         * These values can be obtained by preprocessing the environment map using the <code>cmgen</code> tool of
+         * {@link https://github.com/google/filament/releases | Google's Filament project}. This will also generate a KTX file that can be
+         * supplied to {@link Model#specularEnvironmentMaps}.
+         *
+         * @memberof Model.prototype
+         *
+         * @type {Cartesian3[]}
+         * @default undefined
+         * @demo {@link https://cesiumjs.org/Cesium/Apps/Sandcastle/index.html?src=Image-Based Lighting.html|Sandcastle Image Based Lighting Demo}
+         * @see {@link https://graphics.stanford.edu/papers/envmap/envmap.pdf|An Efficient Representation for Irradiance Environment Maps}
+         */
+        sphericalHarmonicCoefficients : {
```

Comment:
IBL options should be added to `Cesium3DTileset` as well. Should these also be part of both `Model` and `Cesium3DTileset` constructors. It may be worth looking at https://github.com/AnalyticalGraphicsInc/cesium/pull/7172/files#r232321841 sooner than later.

Reply:
It may also be a good idea to add IBL options to the scene level since in most of the time someone will want to use the same environment across the whole scene.